### PR TITLE
Create folder implementation - 3

### PR DESCRIPTION
**Pull request by Tulsishah**
### Description
- Fast stat bucket integration with create folder
- Insert the folder into cache after successfully created.
- Added unit tests.
- Haven't integrated stretcher testify as it will add more changes in this PR.

### Link to the issue in case of a bug fix.
NA

### Testing details
1. Manual - NA
2. Unit tests - Added
3. Integration tests - NA

### CODE REVIEW

File: internal/storage/caching/fast_stat_bucket_test.go [hunk 1]
```diff
@@ -783,3 +784,34 @@ func (t *StatObjectTest) TestRenameFolder() {
 	AssertEq(nil, err)
 	ExpectEq(result, folder)
 }
+
+func (t *StatObjectTest) TestShouldCallCreateFolder() {
```
Comment (vadlakondaswetha) [inline internal/storage/caching/fast_stat_bucket_test.go:788]: TEst method names are not clear. you are testing createFolder and mocking createFolder. Clearly mention which implemetation is throwig error or should be called.

Comment (Tulsishah) [reply internal/storage/caching/fast_stat_bucket_test.go]: Can you please check now?

File: internal/storage/caching/fast_stat_bucket_test.go [hunk 2]
```diff
@@ -812,3 +812,40 @@ func (t *DeleteFolderTest) Test_DeleteFolder_Failure() {
...
 	AssertNe(nil, err)
 }
+
+type CreateFolderTest struct {
+	fastStatBucketTest
+}
+
+func init() { RegisterTestSuite(&CreateFolderTest{}) }
+
+func (t *CreateFolderTest) TestCreateFolderCallsWrappedCreateFolderWithSuccess() {
```
Comment (vadlakondaswetha) [inline internal/storage/caching/fast_stat_bucket_test.go:822]: wrapped is our local variable name. You shouldn't mention all those inn test method names.
Mention what is the scenario you are testinng. Here you are testing what happens when GCS calls succeeds and fails. success/failure are mocked .

TestCreateFolderWhenGCSCallSucceeds

TestCreateFolderWhenGCSCallFails

Comment (Tulsishah) [reply internal/storage/caching/fast_stat_bucket_test.go]: Okay thanks. Done

**Review by vadlakondaswetha — APPROVED** [review]

**Pull request merged**